### PR TITLE
Make param indices of delete_indices method mandatory

### PR DESCRIPTION
**Pull request by wangch079**
## Follow-up on PR https://github.com/elastic/connectors-python/pull/462

This PR
1. makes the param `indices` of method `ESClient.delete_indices` mandatory, 
2. returns early without making ES request if `indices` is empty.

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally by running `make test` and `make ftest NAME=mysql`
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

File: connectors/es/client.py [hunk 1]
```diff
@@ -125,7 +125,8 @@ async def check_exists(self, indices=None, pipelines=None):
             except NotFoundError:
                 raise PreflightCheckError(f"Could not find pipeline {pipeline}")
 
-    async def delete_indices(self, indices=None):
-        if indices is None:
-            indices = []
+    async def delete_indices(self, indices):
+        if indices is None or len(indices) == 0:
```
Comment (tarekziade) [inline connectors/es/client.py:129]: You are still allowing `indices` to be `None` -- do we really want to support a call to `delete_indices` to be a no-op ?

**Review by tarekziade — APPROVED** [review]

**Review by tarekziade — CHANGES_REQUESTED** [review]
the only usage of that method is here: 

https://github.com/elastic/connectors-python/blob/main/connectors/services/job_cleanup.py#L75-L76

where you check that the indices are not empty, so I think `delete_indices` can directly pass the call to `self.client`

File: connectors/es/client.py (hunk 1 above)
Comment (wangch079) [reply connectors/es/client.py:129]: Yes, the caller can still call `an_index.delete_indices(indices=None)`

**Comment by wangch079** [comment]
> where you check that the indices are not empty, so I think `delete_indices` can directly pass the call to `self.client`

Indeed, but I don't think we should define the method based on it's caller's implementation.

File: connectors/es/client.py (hunk 1 above)
Comment (tarekziade) [reply connectors/es/client.py:129]: > > where you check that the indices are not empty, so I think `delete_indices` can directly pass the call to `self.client`
> 
> Indeed, but I don't think we should define the method based on it's caller's implementation.

But we're not implementing an API for hypothetical ways of calling it -- it's added for actual use cases, and right now the one single use case is to delete indices. Why would you call it if you don't intend to delete some indices? what would be the actual use case of passing `None` to it?  I don't see that requirement/need, and I think allowing `None` and silently ignoring it by returning is a recipe for problems.

**Comment by tarekziade** [comment]
> > where you check that the indices are not empty, so I think `delete_indices` can directly pass the call to `self.client`
> 
> Indeed, but I don't think we should define the method based on it's caller's implementation.

But we're not implementing an API for hypothetical ways of calling it -- it's added for actual use cases, and right now the one single use case is to delete indices. Why would you call it if you don't intend to delete some indices? what would be the actual use case of passing None to it? I don't see that requirement/need, and I think allowing None and silently ignoring it by returning is a recipe for problems.

**Comment by wangch079** [comment]
Done here: https://github.com/elastic/connectors-python/pull/489/commits/0e802356e276461378db8ffe8e5a3406f6086fa9

**Review by tarekziade — APPROVED** [review]

**Pull request merged**